spec/system/admin/reports/sales_tax/sales_tax_totals_by_producer_spec.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require 'system_helper'
+
+describe "Sales Tax Totals By Producer" do
+  #  Scenarion 1:
+  #  1 producer
+  #  1 distributor
+  #  1 product that costs 100$
+  #  1 order with 1 line item
+  #  the line item match 2 tax rates: country (2.5%) and state (1.5%)
+  let!(:state_zone){ create(:zone_with_state_member) }
+  let!(:country_zone){ create(:zone_with_member) }
+  let!(:tax_category){ create(:tax_category) }
+  let!(:state_tax_rate){ create(:tax_rate, zone: state_zone, tax_category: tax_category) }
+  let!(:country_tax_rate){ create(:tax_rate, zone: country_zone, tax_category: tax_category) }
+  let!(:ship_address){ create(:ship_address) }
+
+  let!(:variant){ create(:variant) }
+  let!(:product){ variant.product }
+  let!(:supplier){ create(:supplier_enterprise) }
+  let!(:distributor){ create(:distributor_enterprise_with_tax) }
+  let!(:payment_method){ create(:payment_method, :flat_rate) }
+  let!(:shipping_method){ create(:shipping_method, :flat_rate) }
+
+  let!(:order){ create(:order_with_distributor, distributor: distributor) }
+  let!(:order_cycle){
+    create(:simple_order_cycle, suppliers: [supplier], distributors: [distributor],
+                                variants: [variant])
+  }
+
+  let(:admin){ create(:admin_user) }
+
+  before do
+    state_tax_rate.update!({ name: 'State', amount: 0.015 })
+    country_tax_rate.update!({ name: 'Country', amount: 0.025 })
+    tax_category.update!({ name: 'tax_category' })
+
+    distributor.update!({ name: 'Distributor' })
+    distributor.shipping_methods << shipping_method
+    distributor.payment_methods << payment_method
+    supplier.update!({ name: 'Supplier', charges_sales_tax: true })
+
+    product.update!({
+                      tax_category_id: tax_category.id,
+                      supplier_id: supplier.id
+                    })
+    order_cycle.update!(name: "oc1")
+
+    order.line_items.create({ variant: variant, quantity: 1, price: 100 })
+    order.update!({
+                    order_cycle_id: order_cycle.id,
+                    ship_address_id: ship_address.id
+                  })
+
+    while !order.completed?
+      break unless order.next!
+    end
+  end
+
```

Comment:
Hey @abdellani , this looks great :clap:

Some thoughts/suggestions:
This currently tests taxes which are added to the price, I've checked your `state_tax_rate` and `country_tax_rate` and both have `included_in_price: false`. So I'd suggest splitting this spec into two `context`s, one for `added` (done!) and the other for `inclusive` taxes (to do).

For reference, we do this already in some other specs, when testing:
- [invoices](https://github.com/openfoodfoundation/openfoodnetwork/blob/61175c036a221ea5d1638bf378c7639ceb7f6848/spec/system/admin/invoice_print_spec.rb#L137)
- split checkout. Here we've chosen to do this in separate files ([included](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/spec/system/consumer/split_checkout_tax_incl_spec.rb) / [added](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/spec/system/consumer/split_checkout_tax_not_incl_spec.rb)), due to the multiple scenarios, for example, checking out from zones outside the taxed zones. I don't think we need to do this here, though.

As I understand this PR, the report is not calculating any new values (please correct me if I'm wrong!); it is only fetching values already present in in `Spree::Adjustment` table, from `eligible: true` adjustments. So I would say we're good to go, as long as we cover both tax types (added, included). I might be wrong though.

Reply:
Hi @filipefurtad0 

> As I understand this PR, the report is not calculating any new values (please correct me if I'm wrong!); 

Yes, It groups the line items by tax rates and calculates the sum.

---
I'll add a new test for inclusive tax.

Thank you.

Reply:
@filipefurtad0 
I updated the tests.

Reply:
Awesome, @abdellani :clap: :clap: :clap: I have some comments which I'll add below.